edb/edgeql/parser/__init__.py

```diff
@@ -229,24 +228,15 @@ def _cst_to_ast(
             production_id = node.id()
             production = productions[production_id]
 
-            sym = production.lhs.nontermType()
-            assert len(args) == len(production.rhs)
-            production.method(sym, *args)
+            non_term_type, method = production
+            sym = non_term_type()
+            method(sym, *args)
 
             # push into result stack
             result.append(sym)
 
     return result.pop()
 
 
-def preload(allow_rebuild: bool = False) -> None:
-    grammar = qlgrammar.start
-    try:
-        spec = parsing.load_parser_spec(grammar, allow_rebuild=False)
-    except parsing.ParserSpecIncompatibleError as e:
-        if allow_rebuild:
-            spec = parsing.load_parser_spec(grammar, allow_rebuild=True)
-        else:
-            raise errors.InternalServerError(e.args[0]) from None
-
-    rust_parser.cache_spec(spec)
+def preload() -> None:
+    rust_parser.preload_spec()
```

Comment:
Can we keep the hot reload ability for devmode?  It's a lot better if you only have to reload the server as opposed to running `make rust` every time you've changed the parser.

Reply:
It's just `make parsers`, rebuilding Rust is not required.

The problem is that because we don't even init the `parsing.Spec` during runtime anymore, we cannot compare it to find out if it is incompatible.

So we can either rebuild the parsers at every run, or never.

I could port the code from parsing that detects the changes in the spec to the new parser, but it'll take a bit more effort.

Reply:
`make parsers` isn't a thing currently, are you going to add it?